internal/core/operations/schedulers/switchversion/executor.go

```diff
@@ -132,179 +81,14 @@ func (ex *Executor) Execute(ctx context.Context, op *operation.Operation, defini
 		return updateSchedulerErr
 	}
 
-	ex.clearNewCreatedRooms(op.SchedulerName)
 	logger.Info("scheduler update finishes with success")
 	return nil
 }
 
 func (ex *Executor) Rollback(ctx context.Context, op *operation.Operation, definition operations.Definition, executeErr error) error {
-	logger := zap.L().With(
-		zap.String(logs.LogFieldSchedulerName, op.SchedulerName),
-		zap.String(logs.LogFieldOperationDefinition, definition.Name()),
-		zap.String(logs.LogFieldOperationPhase, "Rollback"),
-		zap.String(logs.LogFieldOperationID, op.ID),
-	)
-	logger.Info("starting Rollback routine")
-
-	err := ex.deleteNewCreatedRooms(ctx, logger, op.SchedulerName)
-	ex.clearNewCreatedRooms(op.SchedulerName)
-	if err != nil {
-		logger.Error("error deleting newly created rooms", zap.Error(err))
-		deleteCreatedRoomsErr := fmt.Errorf("error rolling back created rooms: %w", err)
-		ex.operationManager.AppendOperationEventToExecutionHistory(ctx, op, deleteCreatedRoomsErr.Error())
-		return err
-	}
-
-	logger.Info("finished Rollback routine")
```

Comment:
switch version will no longer have rollbacks?

Reply:
No. Previously, the rollback synchronously looped through all newly created rooms and deleted. This generated some issues:

1. Sync operation can take a while to ask for deletion of all game rooms and no operations would run for that scheduler
2. The deletion would not respect any scaling mechanism, thus it would lead to a drastic reduction in the number of game rooms

Now, the switch version can only fail if the update in SQL fails. Since it is ACID, if the update fails then the transaction fails in the DB as well, thus we will not end up in a situation where we need to roll back a transaction manually in the DB.

The following will happen:

1. Operation to switch scheduler version from v1 to v2
2. Change in the DB the active version to v2
3. Next _health_controller_ operation will check that we have a rolling update - still exists pods in the non-active versions
4. _health_controller_ performs rolling update adding v2 and deleting v1 pods
5. At some point, one wants to roll back to v1 and will send an API command to switch the scheduler version. This will be translated into the switch version operation to the worker for v1
6. _switch_version_ will change in the DB the version back to v1
7. Next, the _health_controller_ operation detects an ongoing rolling update to v1
8. _health_controller_ performs rolling update adding v1 and deleting v2 pods

This is in line with how Kubernetes does. Rolling back a version triggers a new deployment and the system in each iteration can read its current state and determine what actions to take, rather than trying to control the full lifecycle of the operation in one batch as we were doing on the previous _switch_version_